Lib/asyncio/selector_events.py

```diff
@@ -1113,6 +1167,16 @@ def write_eof(self):
         if not self._buffer:
             self._sock.shutdown(socket.SHUT_WR)
 
+    def writelines(self, list_of_data):
+        if self._eof:
+            raise RuntimeError('Cannot call writelines() after write_eof()')
+        if self._empty_waiter is not None:
+            raise RuntimeError('unable to writelines; sendfile is in progress')
+        if not list_of_data:
+            return
+        self._buffer.extend([memoryview(data) for data in list_of_data])
+        self._write_ready()
```

Comment:
@kumaraditya303 should this have instead been `self._loop._add_writer(self._sock_fd, self._write_ready)` since without it there will be no writer to clear the buffer in case `sendmsg` is not able to clear the entire buffer in one go.